### ISSUE TITLE
[Bug] Channel mismatch in decoder upsample layer causes RuntimeError

### ISSUE THREAD

**Issue by SoupJava** [va1shn9v/PromptIR#48]
## 🐛 Bug Description
When running the model with `decoder=True`, a `RuntimeError` occurs during the forward pass of the decoder.

### Root Cause
The mismatch happens at `self.up4_3(latent)` in `PromptIR.forward`.

- `self.latent` (bottleneck) outputs **384 channels** (since `Downsample` doubles the channels: `dim*2**3 = 384`).
- `self.up4_3` is initialized with `Upsample(int(dim*2**2))`, which expects **192 input channels**.
- Thus the first convolution inside `Upsample` receives 384 channels but expects 192.

My Proposed Fix
Change two lines in PromptIR.__init__:

# Line ~160, change Upsample input channels from 192 to 384
self.up4_3 = Upsample(int(dim*2**3))   # was: Upsample(int(dim*2**2))

# Line ~162, adjust reduce_chan_level3 input channels accordingly
# After up4_3 output (192) and out_enc_level3 (192) are concatenated → total 384
self.reduce_chan_level3 = nn.Conv2d(int(dim*2**3), int(dim*2**2), kernel_size=1, bias=bias)

# I want to know what the value of this part of the previously successfully trained model is. Is it a different value?

**Comment by Amazingren** [comment]
您的来信已经收到，稍后进行回复，谢谢~

**Comment by SoupJava** [comment]
Resolved
